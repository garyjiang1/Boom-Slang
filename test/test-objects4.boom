

class MyClass:
	required:
		MyClass myclass
		int x

	def construct():
		# do nothing
		2 + 2

	def getself() returns MyClass:
		return self

	def printx():
		println(self.x)

	def to_string() returns string:
		return "this is my custom to string"



MyClass innermost = MyClass()
MyClass foo = MyClass(MyClass(MyClass(MyClass(MyClass(MyClass(innermost, 1), 2), 3), 4), 5), 6)
println(foo)
println(foo.myclass.myclass.getself().getself().getself().myclass.x)
println(foo.getself().getself().x)
println(foo.getself().myclass.myclass.getself().myclass)
println(foo.myclass.myclass)
foo.getself().getself().printx()

println(MyClass())
println(MyClass().x)
MyClass().printx()
println(MyClass().to_string())
