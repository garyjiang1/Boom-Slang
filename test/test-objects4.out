this is my custom to string
3
6
this is my custom to string
this is my custom to string
6
this is my custom to string
0
0
this is my custom to string
